[zh-cn]: update translation of Map.@@iterator() (#16444)
mdn/translated-content: files/zh-cn/web/javascript/reference/global_objects/map/@@iterator/index.md

{{JSRef}}

> {{jsxref("Map")}} 实例的 **`[@@iterator]()`** 方法实现了[可迭代协议](/zh-CN/docs/Web/JavaScript/Reference/Iteration_protocols)以允许 `Map` 对象被大多数语法所接受，例如 [展开语法](/zh-CN/docs/Web/JavaScript/Reference/Operators/Spread_syntax)和 {{jsxref("Statements/for...of", "for...of")}} 循环。它返回一个 [map 迭代器对象](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Iterator) 它会以插入顺序返回 map 的键值对。

Reviewer: ```suggestion
{{jsxref("Map")}} 实例的 **`[@@iterator]()`** 方法实现了[可迭代协议](/zh-CN/docs/Web/JavaScript/Reference/Iteration_protocols)以允许 `Map` 对象被大多数语法所接受，例如[展开语法](/zh-CN/docs/Web/JavaScript/Reference/Operators/Spread_syntax)和 {{jsxref("Statements/for...of", "for...of")}} 循环。它返回一个 [map 迭代器对象](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Iterator)，此对象会以插入顺序生成 map 的键值对。
```

该属性的初始值与 {{jsxref("Map.prototype.entries")}} 属性的初始值是同一个函数对象。

{{EmbedInteractiveExample("pages/js/map-prototype-@@iterator.html")}}

## 语法

```js-nolint
map[Symbol.iterator]()
```

### 参数

无。

### 返回值

> 与 {{jsxref("Map.prototype.entries()")}} 返回值相同，一个新的[迭代器对象](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Iterator)，它会以插入顺序返回 map 的键值对。

Reviewer: ```suggestion
与 {{jsxref("Map.prototype.entries()")}} 返回值相同：一个新的[迭代器对象](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Iterator)，它会以插入顺序生成 map 的键值对。
```

## 示例

### 使用 for...of 循环进行迭代

> 请注意，通常你不需要直接调用此方法。`@@iterator` 方法的存在使得 `Map` 对象[可迭代](/zh-CN/docs/Web/JavaScript/Reference/Iteration_protocols#可迭代协议)，而迭代语法如 `for...of` 循环会自动调用此方法以获取用于循环的迭代器。

Reviewer: ```suggestion
请注意，通常你不需要直接调用此方法。`@@iterator` 方法的存在使得 `Map` 对象[可迭代](/zh-CN/docs/Web/JavaScript/Reference/Iteration_protocols#可迭代协议)，而像 `for...of` 循环这样的迭代语法会自动调用此方法以获取用于循环的迭代器。
```
